lib/util/Components.js

```diff
@@ -215,7 +215,14 @@ function componentRule(rule, context) {
      * @returns {Boolean} True if the node is explicitly declared as a descendant of a React Component, false if not
      */
     isExplicitComponent: function(node) {
-      const comment = sourceCode.getJSDocComment(node);
+      let comment;
+      // Sometimes the passed node may not have been parsed yet by eslint, and this function call crashes.
+      // Can be removed when eslint sets "parent" property for all nodes on initial AST traversal: https://github.com/eslint/eslint-scope/issues/27
```

Comment:
It does not help. The `relatedComponent` (in the case below it is `MyComponent`) is not a child in the AST of the `MemberExpression` `MyComponent.PROPTYPES` - so using `:exit` does not work.

The reason is just that `MyComponent` is declared after `MyComponent.PROPTYPES` and therefore eslint has not yet parsed it and added the `parent` property.

```js
      MyComponent.PROPTYPES = {}
      /** @extends React.Component */
      class MyComponent extends BaseComponent {}
``` 

They have a task to add parent property to all nodes on initial traversal, that would fix our issue.